app/scripts/controllers/transactions/index.js

```diff
@@ -1996,6 +1996,13 @@ export default class TransactionController extends EventEmitter {
       if (otherTxMeta.id === txId) {
         return;
       }
+      // We don't want to mark as dropped any transactions that are originally created by this account and are incoming
+      if (
+        otherTxMeta.type === TransactionType.incoming &&
```

Comment:
Rather than adding multiple early returns inside this loop, shall we just add a `filter` to the `sameNonceTxs` above to ignore those with the same ID or with the `incoming` type?

Reply:
Done